Contributing.md

```diff
@@ -56,6 +56,70 @@ discuss pending contributions in order to find a resolution. It is expected that
+express repos, but emanating from the usage of it are classified as `question`
+
+`bug`: Issues that, after investigation, provides a reasonable conviction about
+a potential bug in the express code, are classified as `bug`
+
+`enhancement`: Issues that, after discussion, investigation, and technical assessment,
+found to be a reasonable candidate for a good to have feature, are tagged as `enhancement`.
+
+Raisers may appeal to change the tag with sufficient reasoning.
+Maintainers can change the tag based on the change in inference in the course of investigation.
+
+In all cases, issues may be closed by maintainers if they don't receive timely response when
+further information is sought, or when additional questions are asked.
+
+## Approaches and Best Practices for getting into triage contributions
+
+ - review the organization's [Dashboard][]
+ - pay special attention to these columns: stars, watchers, open issues and contributors
+ - this gives you a general idea about the criticality and health of the repository
+ - pick choose the top n based on that criteria
+ - augment that info with your skills in the middleware space - existing or aspiring
+ - refine the priority based on that
+ - for a specific repo, do the following:
+   - review the project's contribution guideline if present. In a nutshell, pledge yourself to the community's standards and values.
+   - review the documentation. for most of the projects, it is a single page doc
+   - understand the key functions: APIs, their semantics, configurations and use cases
+   - write `own` test programs to re-affirm your understanding on these functions
+   - this `may` throw up some gaps in documentation. record those gaps
+   - skim through the issue backlog; identify low hanging ones - mostly new ones
+   - attempt to recreate issues based on the OP description. ask questions if required
+   - validate with the API spec / doc.
+   - it is a good idea to turn on DEBUG to get the next level information
+   - it is also a good idea to do live debugging to follow the control flow
+   - review the full list of issue backload, and make some categorization convenient to your way of work:
+     - new, unattended issues
+     - simple questions
+     - hard bugs that are unresolved for years
+     - ongoing discussions on APIs and functions
+     - issues that imply potential RFEs
+     - issues that imply gaps in documentation
+     - issues that imply recurring patterns
+     - these are not necessarily a disjoint list, mostly can have overlaps
+  - start addressing one type at a time - that build up knowledge on that area of code, interface and configuration
+  - at this point it is a good idea to make one pass of reading through the entire source
+    - this may be possible for smaller components like session, but may be tough for big ones (like express)
+  - for recurring patterns, it is a good idea to demonstrate through fully functional programs and publish through gist for later re-use
```

Comment:
probably typo ? - 

publish through gist --> publish thorough gist

Reply:
@samkamer - here I meant `through` itself; gist would mean https://gist.github.com/ , where we could publish re-usable artifacts

Reply:
Ah, got it ! Then it is good as is.